### ISSUE TITLE
Improve `UnnecessaryFilter`

### ISSUE THREAD

**Issue by BraisGabin** [detekt/detekt#6429]
## Expected Behavior of the rule
Code like this should be flagged by `UnnecessaryFilter`: 
```kotlin
val smells = filterAutoCorrectedIssues(config)
  .flatMap { it.value }
  .filter { it.severity == Severity.ERROR }
return smells.count()
```

As it is the same as 

```kotlin
return filterAutoCorrectedIssues(config)
  .flatMap { it.value }
  .count { it.severity == Severity.ERROR } 
```

## Context
We had this issue on our own code: #6409.
